tests/integration/test_eoc_journal.py

```diff
@@ -585,32 +602,108 @@ def test_takeaways_pdf_enabled_and_configured(self):
         link = self.element.find_element_by_css_selector('a.key-takeaways-link')
         self.assertIn('Key Takeaways', link.text)
         self.assertEqual(link.get_attribute('target'), '_blank')
+        self.assert_in_student_view_data('display_key_takeaways', True)
+        self.assert_in_student_view_data('key_takeaways_pdf_url', '/static/my.pdf')
 
     def test_no_pdf_report_link_heading_configured(self):
         selected_block_ids = default_pb_answer_block_ids[1:]
         self.configure_block(selected_pb_answer_blocks=selected_block_ids)
         heading = self.element.find_element_by_css_selector('.eoc-pdf-report .title')
         self.assertEqual('PDF Report', heading.text)
+        self.assert_in_student_view_data('pdf_report_link_heading', 'PDF Report')
 
     def test_pdf_report_link_heading_configured(self):
         selected_block_ids = default_pb_answer_block_ids[1:]
         self.configure_block(selected_pb_answer_blocks=selected_block_ids,
                              pdf_report_link_heading='My Report')
         heading = self.element.find_element_by_css_selector('.eoc-pdf-report .title')
         self.assertEqual('My Report', heading.text)
+        self.assert_in_student_view_data('pdf_report_link_heading', 'My Report')
 
     def test_no_pdf_report_link_text_configured(self):
         selected_block_ids = default_pb_answer_block_ids[1:]
         self.configure_block(selected_pb_answer_blocks=selected_block_ids)
         link = self.element.find_element_by_css_selector('a.pdf-report-link')
         self.assertEqual('Download PDF', link.text)
+        self.assert_in_student_view_data('pdf_report_link_text', 'Download PDF')
 
     def test_pdf_report_link_text_configured(self):
         selected_block_ids = default_pb_answer_block_ids[1:]
         self.configure_block(selected_pb_answer_blocks=selected_block_ids,
                              pdf_report_link_text='Download Report')
         link = self.element.find_element_by_css_selector('a.pdf-report-link')
         self.assertEqual('Download Report', link.text)
+        self.assert_in_student_view_data('pdf_report_link_text', 'Download Report')
+
+    def test_student_view_data(self):
+        selected_block_ids = default_pb_answer_block_ids[1:]
+        self.configure_block(selected_pb_answer_blocks=selected_block_ids, )
+        block = self.load_root_xblock()
+        data = block.student_view_data()
+        self.assertEqual(data, {
+            'display_name': u'Course Journal',
+            'display_key_takeaways': False,
+            'pdf_report_link_heading': u'PDF Report',
+            'pdf_report_url': u'/handler/test-scenario.eoc-journal.d0.u0/serve_pdf/?student=student_1',
```

Comment:
I think key takeaways is already absolute via _expand_static_url.

Reply:
@xitij2000 Thanks for fixing the `pdf_report_url` -- it's showing up as absolute now (when I re-publish the course to refresh the course blocks cache).

But sorry, no, the  `key_takeways_url` isn't being made absolute, since it [calls `_expand_static_url`](https://github.com/open-craft/xblock-eoc-journal/blob/2881638b919d21c0441ad6dfd248c933683c91fe/eoc_journal/eoc_journal.py#L173) with the default `absolute=False` argument.